pkg/controllers/dynakube/controller.go

```diff
@@ -297,23 +301,15 @@ func (controller *Controller) setupTokensAndClient(ctx context.Context, dynakube
 
 	controller.setConditionTokenReady(dynakube)
 
-	log.Info("start reconciling pull secret")
-
-	err = dtpullsecret.
-		NewReconciler(controller.client, controller.apiReader, dynakube, tokens).
-		Reconcile(ctx)
-	if err != nil {
-		log.Info("could not reconcile Dynatrace pull secret")
-
-		return nil, err
-	}
-
 	return dynatraceClient, nil
 }
 
 func (controller *Controller) reconcileComponents(ctx context.Context, dynatraceClient dtclient.Client, istioClient *istio.Client, dynakube *dynatracev1beta1.DynaKube) error {
 	var componentErrors []error
 
+	pullSecretReconciler := dtpullsecret.
+		NewReconciler(controller.client, controller.apiReader, dynakube, controller.tokens)
+
```

Comment:
why is this change necessary for the `using the tennantUUID from the connection-info` logic?

Also, it looks kinda wrong (sorry if its a bit rambly 😅 )
- when do you **not** need a pull-secret? => when its application-monitoring
   - for both oneagent and activegate it is needed
   - so why move it into the `injection` reconciler?
   - why move it at all in the context of the intended change?
 - why store a "status" in memory that will be released right after each reconcile?
   - The current way we do it for the`connectionInfo` is that the `injection` and `oneagent` Reconcilers don't each get a `connection-info-reconciler` but create one on there own, without coupling everything together.
      - The "is already reconciled" logic is done using the status and the timestamp within it, so its multipurpose
         - Only gets updated every X minutes
         - Only get updated maximum once per reconcile as the timestamp already handles that

Reply:
You are right, the creation does not need to be in the injection reconciler - Added it in the injection reconciler, as I was not sure if the oneagent reconciler will always run before the injection one. But guess this won't change that likely in the future. Will remove it from there. 

The Status is way more elegant did not think about it, thx for pointing out 👍